packages/grid/x-data-grid-pro/src/components/DataGridProColumnHeaders.tsx

```diff
@@ -81,7 +81,7 @@ const GridColumnHeadersPinnedColumnHeaders = styled('div', {
   display: 'flex',
   flexDirection: 'column',
   boxShadow: theme.shadows[2],
-  backgroundColor: theme.palette.background.default,
+  backgroundColor: (theme.vars || theme).palette.background.default,
   ...(theme.palette.mode === 'dark' && {
     backgroundImage: `linear-gradient(${alpha('#fff', getOverlayAlpha(2))}, ${alpha(
```

Comment:
It is the same function as https://github.com/siriwatknp/material-ui/blob/system/use-css-theme-vars/packages/mui-material/src/styles/experimental_extendTheme.js#L13, so using `theme.vars.overlays[2]` should work the same.

Reply:
Yes, sounds to already be the case for Paper component

https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Paper/Paper.js#L53-L62